vsphere/tests/test_check.py

```diff
@@ -431,3 +432,34 @@ def test_specs_start_time(aggregator, dd_run_check, historical_instance):
     assert len(start_times) != 0
     for start_time in start_times:
         assert start_time == (mock_time - dt.timedelta(hours=2))
+
+
+@pytest.mark.parametrize(
+    'test_timeout, expected_result',
+    [
+        (5, False),
+        (10, False),
+        (20, True),
+        ('5', False),
+        ('20', True),
+    ],
+)
+@pytest.mark.usefixtures('mock_type', 'mock_threadpool', 'mock_api', 'mock_rest_api')
+def test_connection_refresh(aggregator, dd_run_check, realtime_instance, test_timeout, expected_result):
+    # This test is to ensure that the connection is refreshed after a specified period of time.
+    # We run the check initially to get a connection object, sleep for a period of time, and then
+    # rerun the check and compare and see if the connection objects are the same.
+    realtime_instance['connection_reset_timeout'] = test_timeout
+    check = VSphereCheck('vsphere', {}, [realtime_instance])
+    dd_run_check(check)
+    first_connection = check.api
```

Comment:
Another way to check the number of times a new connection is made is by using a mock: https://stackoverflow.com/questions/57768709/counting-calls-to-method-inside-method-in-python-unittest